cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogRecoveryCommand.java

```diff
@@ -91,55 +85,43 @@
     public DurableLogRecoveryCommand(CommandArgs args) {
         super(args);
         this.containerCount = getServiceConfig().getContainerCount();
-        this.storageFactory = createStorageFactory(executorService);
-        this.storage = this.storageFactory.createStorageAdapter();
-        this.zkClient = createZKClient();
+        this.storageFactory = createStorageFactory(this.executorService);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        @Cleanup
+        Storage storage = this.storageFactory.createStorageAdapter();
+        @Cleanup
+        val zkClient = createZKClient();
 
         val bkConfig = getCommandArgs().getState().getConfigBuilder()
                 .include(BookKeeperConfig.builder().with(BookKeeperConfig.ZK_ADDRESS, getServiceConfig().getZkURL()))
                 .build().getConfig(BookKeeperConfig::builder);
-        this.dataLogFactory = new BookKeeperLogFactory(bkConfig, this.zkClient, executorService);
-    }
-
-    @Override
-    public void close() throws Exception {
-        if (this.dataLogFactory != null) {
-            this.dataLogFactory.close();
-        }
-        this.zkClient.close();
-        this.storage.close();
-        ExecutorServiceHelpers.shutdown(Duration.ofSeconds(2), executorService);
-    }
+        @Cleanup
+        val dataLogFactory = new BookKeeperLogFactory(bkConfig, zkClient, executorService);
 
-    @Override
-    public void execute() throws Exception {
         outputInfo("Container Count = %d", this.containerCount);
 
-        try {
-            this.dataLogFactory.initialize();
-        } catch (DurableDataLogException ex) {
```

Comment:
`zkClient` and `dataLogFactory` need not to be closed, as they are initialized with `@CleanUp` tag. In case of an exception, it will popped up and `cleanUp` will clan these variables.